docs/how-to-work-on-coding-challenges.md

```diff
@@ -31,96 +31,135 @@ Before you work on the curriculum, you would need to set up some tooling to help
 
 ## Challenge Template
 
-Below is a template of what the challenge markdown files look like currently.  To see the streamlined template we will be adopting see [below](#upcoming-challenge-template).
-
 ````md
+
 ---
 id: Unique identifier (alphanumerical, MongoDB_id)
-title: Challenge Title
-challengeType: 0
+title: 'Challenge Title'
+challengeType: Integer, defined in `client/utils/challengeTypes.js`
 videoUrl: 'url of video explanation'
+forumTopicId: 12345
 ---
 
-## Description
+# --description--
 
-<section id='description'>
-A Description of the challenge and what is required to pass
-</section>
+Challenge description text, in markdown
 
-## Instructions
+```html
+<div>
+  example code
+</div>
+```
 
-<section id='instructions'>
-Instructions about what exactly needs to be done.
-</section>
+# --instructions--
 
-## Tests
+Challenge instruction text, in markdown
 
-<section id='tests'>
+# --hints--
 
-```yml
-tests:
-  - text: Should return "foo"
-    testString: 'A stringified function possibly using Chai asserts'
+Tests to run against user code, in pairs of markdown text and codeblock test code.
+
+```js
+Code for test one
 ```
 
-</section>
+More instructions in markdown syntax
 
-## Challenge Seed
+```js
+More code
+```
 
-<section id='challengeSeed'>
+# --seed--
+
+## --before-user-code--
 
-<div id='{ext}-seed'>
+```lang
+Code evaluated before the user’s code.
+```
 
-```{ext}
-Code displayed in the editor by default.
+## --after-user-code--
 
-This is a required section for the challenge.
+```lang
+Code evaluated after the user’s code, and just before the tests
 ```
 
-</div>
+## --seed-contents--
+
+Boilerplate code to render to the editor
 
-### Before Test
+```html
+<body>
+  <p class="main-text">
+    Hello world!
+  </p>
+</body>
+```
+
+```css
+body {
+  margin: 0;
+  background-color: #3a3240;
+}
 
-<div id='{ext}-setup'>
+.main-text {
+  color: #aea8d3;
+}
+```
 
-```{ext}
-Optional Test setup code.
+```js
+console.log('freeCodeCamp is awesome!');
 ```
 
-</div>
+# --solutions--
 
-### After Test
+Solutions are used for the CI tests to ensure that changes to the hints will still pass as intended
 
-<div id='{ext}-teardown'>
+```js
+// first solution
+```
```

Comment:
I can't edit them in, but the solutions need to be separated by `---`, same as answers for video questions.

Also, could we say something like

> // first solution, the language(s) should match the seed

Reply:
> // first solution, the language(s) should match the seed

Just for clarity sake, this is to mean that the language used for the codeblock (i.e. `js`, `html`, etc) should be the same for the seed and the solution(s) correct?

Reply:
Right, the codeblock language. If there are multiple seed languages then each solution language should match one of the ones in the seed.

Reply:
Alright - I've clarified this in the template. 🙂